### PR TITLE
[docs] Add sorting page in datagrid docs

### LINKED ISSUE #925: [docs] Move sorting into its own page
## Summary 💡

The sorting feature is one of the most popular features of a data grid, at least, from a community perspective. Check the ranking usage of react-table between the different demos (at a giving point in time):

- baseline 135.6k
- filtering 45.8k
- pagination 34.6k
- **sorting** 37.6k
- cell edition 28.4k
- row checkbox 26.4k
- row expand 20.6k
- grouping 19.0k
- column resizing 11.9k
- row drag and drop 10.7k
- column order 10.5k
- footer 8.7k
- column hidden 6.0k

We should make it more visible, it should be part of the side navigation

## Motivation 🔦

In its current configuration, it's too challenging to find this page https://material-ui.com/components/data-grid/rows/#row-sorting. Having a link in the side navigation, either as a child of "Rows" or as a flat item until we need to break down "Rows" int o multiple pages will help.

### PR DESCRIPTION
**Pull request by dtassone**
fix #925

### CODE REVIEW

File: docs/src/pages.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ datagrid.children =
         { pathname: '/components/data-grid/getting-started' },
         { pathname: '/components/data-grid/columns' },
         { pathname: '/components/data-grid/rows' },
+        { pathname: '/components/data-grid/sorting', title: 'Sorting' },
         { pathname: '/components/data-grid/filtering', title: 'Filtering' },
```
Comment (oliviertassinari) [inline docs/src/pages.js:32]: Filtering is more popular #925. Why include a `title`?

```suggestion
        { pathname: '/components/data-grid/filtering' },
        { pathname: '/components/data-grid/sorting' },
```

x2

**Comment by dtassone** [comment]
🤔 
image

**Review by oliviertassinari — CHANGES_REQUESTED** [review]
- We don't allow two h1 on a page
- Please link the issue it's fixing

**Review by oliviertassinari — APPROVED** [review]
Mind that the navigation will only impact the content hosted at https://master--material-ui-x.netlify.app/. We also need to update https://master--material-ui.netlify.app/ to get a consistent UX.

**Pull request merged**